README.md
# mb.imgNavigator 2.0

__An open source jQuery component to easily create a navigator tool for your extralarge images.__

![mb.imgNavigator](http://pupunzi.com/gitHub/mb.imgNavigator.jpg)


## [go to the demo](http://pupunzi.com/#mb.components/mb.imageNavigator/imageNavigator.html)
## [go to the doc](http://wiki.github.com/pupunzi/jquery.mb.imgNavigator/)
## [go to the blog](http://pupunzi.open-lab.com/mb-jquery-components/mb-imagenavigator/)


[jquery.mb.components](http://pupunzi.com/), another way of thinking the web


inc/changeLog.txt
jquery.mb.imageNavigator

2.0 Major issue released:
  Solved the containment issue; now the image never goes out from the screen!
  It needs the jQuery UI 1.7.1

1.8.0 Major issue released:
  Now yo can define paramiters without writing custom attributes on the tag, and the component is W3C compliant!:
  >> before
  <div imageUrl="iStock_2.jpg" navPosition="TR" navWidth="100">
  >> Now
  <div style="display:none;">
  You can still use both sintax, but if you want to be w3c compliant than you nee the second one and you have to include jquery.metadata.js in your page.
  you can add an additional content layer that overlay image and moves with it:
  Ex:
  <div style='color:white; font-size:100px; top:0; width:1000px; height:1000px; text-align:left;'>
  this is an additional content div that is over the image and it moves with it!
  </div>


inc/mbImgNav.pack.js
/*******************************************************************************
 jquery.mb.components
 Copyright (c) 2001-2010. Matteo Bicocchi (Pupunzi); Open lab srl, Firenze - Italy
 email: info@pupunzi.com
 site: http://pupunzi.com

 Licences: MIT, GPL
 http://www.opensource.org/licenses/mit-license.php
 http://www.gnu.org/licenses/gpl.html
 ******************************************************************************/

eval(function(p,a,c,k,e,r){e=function(c){return(c<a?'':e(parseInt(c/a)))+((c=c%a)>35?String.fromCharCode(c+29):c.toString(36))};if(!''.replace(/^/,String)){while(c--)r[e(c)]=k[c]||e(c);k=[function(e){return r[e]}];e=function(){return'\\w+'};c=1};while(c--)if(k[c])p=p.replace(new RegExp('\\b'+e(c)+'\\b','g'),k[c]);return p}('(7($){1v.1M.2q=7(s){10 z.1f(7(){z.L={1g:11,1N:11,1w:2r,1O:"2s 2t 2u",1P:.8,1Q:"1R.2v",2w:""};$.1S(z.L,s);s=z.L;A c=z;A d,9,Q,D,Y,M,v,N,R,B,V,1a,1h,w,W,Z,1T,1x,X,16=0,1i=1j,1y,1z;$(c).6({G:c.L.1g});A f=$(c).K(".2x");$(c).2y();C($.S){$.S.2z("T");$(f).1f(7(){C($(z).S().1b)$(z).E("1b",$(z).S().1b);C($(z).S().1c)$(z).E("1c",$(z).S().1c);C($(z).S().12)$(z).E("12",$(z).S().12);C($(z).S().13)$(z).E("13",$(z).S().13)})}A g=$(f[0]).E("1b");A h=$(f[0]).E("1c");A j=$(f[0]).E("12")?$(z).E("12"):c.L.1w;A k=$(f).K(".2A");A m=$(f).K(".2B");A n=$(f).K(".1A");$(c).O("<I T=\'v\'></I>");v=$(c).K(".v");$(v).6({2C:"2D",P:"1B",G:z.L.1g+"1U",J:z.L.1N+"1U"});A o="<1V 1W=\'1C\' 1X=\'2E:2F;\' 2G=\'0\' 2H=\'0\' G=\'1Y%\' J=\'1Y%\'><1Z><20 2I=\'2J\' 2K=\'2L\'><2M 17=\'"+c.L.1Q+"\' 2N=\'1R\'></20></1Z></1V>";$(v).O(o);$(c).2O("<I T=\'21\'></I>");A p=$(c).K(".21");$(p).6({P:"1B",G:z.L.1g,2P:"0"});$(p).O(k);A q=1k;$(k).1f(7(i){$(z).1D(7(){C(!q)10;16=i;q=1j;g=$(f[i]).E("1b");h=$(f[i]).E("1c");j=$(f[i]).E("12")?$(f[i]).E("12"):c.L.1w;C($(f[i]).E("13")){1y=$(f[i]).E("13").22(",")[0];1z=$(f[i]).E("13").22(",")[1]}C(D)$(D).23(11,7(){1l(g)});U 1l(g)})});$(c).O("<I T=\'24\'></I>");A r=$(c).K(".24");$(r).1E(m[0]);7 1l(u){W=1F;X=1F;C(Q){$(Q).2Q()}$(k).1f(7(i){C(i==16)$(z).2R("25");U $(z).2S("25")});W=j;A a=2T 2U();a.17=1F;u=u+"?2V="+2W.2X();a.17=u;a.2Y=7(){26(u)};a.2Z=27;$("#1C").1G(11)}7 27(){30("31 32 a 33: "+z.17)}7 26(u){N=$(v).G();R=$(v).J();$("#1C").23(11,7(){q=1k});$(k).18("1D",7(){10 1k});$(v).1D(7(){C($.14.15)$(w).1d();U $(w).1d()});$(v).34(7(){C($.14.15)$(w).19();U $(w).19()});$(v).O("<I T=\'Q\'></I>");Q=$(c).K(".Q");$(Q).6({P:"1B",J:$(v).J()});$(Q).O("<I T=\'9\'></I>");9=$(Q).K(".9");$(9).O("<D T=\'28\'>");D=$(9).K(".28");$(9).O("<I T=\'1A\'></I>");d=$(9).K(".1A");$(d).6({P:"1m",F:"0"});$(9).6({P:"1m"});$(r).1E(m[16]);$(d).1E(n[16]);$(D).E("17",u);$(D).19();M=$(D).29();Y=$(D).1H();$(D).1G(35,7(){A t,l;C(!$(f[16]).E("13")){t=-(M/2-($(w).J()*3));l=-(Y/2-($(w).G()*3))}U{t=-(1z-($(v).J()/2));l=-(1y-($(v).G()/2));C(t>0)t=0;C(l>0)l=0}$(9).36({F:t,H:l},11,"37");1I(l,t)});$(9).2a({2b:[$(v).1n().H-Y+$(v).1H(),$(v).1n().F-M+$(v).29(),$(v).1n().H,$(v).1n().F],2c:7(){$(w).19();$(9).6({1o:"2d"})},2e:7(e,a){1I(a.P.H,a.P.F);$(9).6({1o:"1J"});$(w).1d()}});$(9).18("1K",7(){1p()});$(Q).O("<I T=\'w\'></I>");w=$(c).K(".w");$(w).6({P:"1m",2f:c.L.1P});$(w).O("<I 1W=\'B\'></I>");B=$(c).K("#B");$(B).6({38:39,P:"1m",3a:c.L.1O,3b:$.14.15?"3c":"3d",2f:$.14.15?.5:1});$(w).19(1);$(B).18("1K",7(){1p()});$(B).2a({2b:\'3e\',2c:7(){$(B).6({1o:"2d"})},3f:7(e,a){1L(a.P.H,a.P.F)},2e:7(){$(B).6({1o:"1J"})}});$(w).O("<D T=\'V\'>");V=$(c).K(".V");$(V).E("17",u);$(V).18("1K",7(){1p()});1T=M<Y;1x=R<N;X=M/Z;1q()}7 1p(){C(!1i){B.1r=$(B).6("H");B.1s=$(B).6("F");9.1r=$(9).6("H");9.1s=$(9).6("F");A a=((W*R)/Z)<N;C(a)$(D).6("G",N);U $(D).6("J",R);1i=1k;$(B).6("F",0);$(B).6("H",0);$(9).6("F",0);$(9).6("H",0);$(9).18("2g",c.2h=7(){10 1j});$(d).19()}U{$(D).G("");$(D).J("");1i=1j;$(9).6("F",9.1s);$(9).6("H",9.1r);$(B).6("F",B.1s);$(B).6("H",B.1r);$(9).3g("2g",c.2h);$(d).1d()}1q();A x=$(B).3h;A y=$(B).3i;1L(x,y)}7 1L(x,y){X=M/Z;A a=-(1t[0]+1)*X;A b=-(1t[1]+1)*X;$(9).6("F",b);$(9).6("H",a)}7 1I(x,y){X=M/Z;A a=-(1t[0]+1)/X;A b=-(1t[1]+1)/X;$(B).6({F:b,H:a})}7 2i(){W=!W?N/4:W;Z=(W*M)/Y;$(V).J(3j(Z));1a=$(V).G();1h=$(V).J()}7 2j(){$(B).G((N*W)/Y);$(B).J((R*Z)/M)}7 2k(){3k(h){1u"3l":$(w).6("H",0);$(w).6("F",0);1e;1u"3m":$(w).6("F",0);$(w).6("H",(N-1a));1e;1u"3n":$(w).6("F",(R-1h));1e;1u"3o":$(w).6("H",(N-1a));$(w).6("F",(R-1h));1e;1J:A a=($(v).G())-1a;$(w).6("H",a);1e}}7 1q(){N=$(v).G();R=$(v).J();1x=R<N;M=$(D).J();Y=$(D).G();2i();2j();2k()}7 2l(){C(!D)10;$(D).G("");$(D).J("");C($.14.15)$(w).1d();U $(w).1G(11);v.3p=$(v).6("G");v.3q=$(v).6("J");v.1X.G=$(2m).1H();$(2m).18("3r",7(){2l()});1q()}1l(g)})};1v.1M.1S({3s:7(e){A a;C($.14.15){a=2n.3t+2o.2p.3u}U{a=e.3v}C(a<0){a=0}10 a},3w:7(e){A a;C($.14.15){a=2n.3x+2o.2p.3y}U{a=e.3z}C(a<0){a=0}10 a}})})(1v);',62,222,'||||||css|function||draggableElement||||||||||||||||||||||imageContainer|nav|||this|var|navLocator|if|image|attr|top|width|left|div|height|find|options|imageH|imageContainerW|append|position|applContainer|imageContainerH|metadata|class|else|navigationThumb|navW|ratio|imageW|navH|return|500|navWidth|NavCoordinates|browser|msie|actualIdx|src|bind|hide|navigationThumbW|imageUrl|navPosition|show|break|each|areaWidth|navigationThumbH|onScreen|false|true|startNav|absolute|offset|cursor|fitonScreen|resetAllValue|oldX|oldY|arguments|case|jQuery|defaultnavWidth|cont_isHoriz|navCoordinateX|navCoordinateY|additionalContent|relative|loader|click|html|null|fadeIn|outerWidth|refreshThumbPos|default|dblclick|refreshImagePos|fn|areaHeight|draggerStyle|navOpacity|loaderUrl|loading|extend|image_isHoriz|px|table|id|style|100|tr|td|imagesIndex|split|fadeOut|descriptionBox|selected|buildnav|imageFailed|navImage|outerHeight|draggable|containment|start|move|stop|opacity|mousemove|doNothing|setnavDim|setnavLocatorDim|setnavPos|fullScreen|window|event|document|body|imageNavigator|150|2px|dotted|red|gif|additionalContenet|imagesContainer|empty|setType|title|description|overflow|hidden|display|none|cellpadding|cellspacing|valign|middle|align|center|img|alt|prepend|padding|remove|addClass|removeClass|new|Image|rdm|Math|random|onload|onerror|alert|non|riesco|caricare|mouseleave|1000|animate|linear|zIndex|10000|border|backgroundColor|white|transparent|parent|drag|unbind|offsetLeft|offsetTop|parseFloat|switch|TL|TR|BL|BR|oldW|oldH|resize|getMouseX|clientX|scrollLeft|pageX|getMouseY|clientY|scrollTop|pageY'.split('|'),0,{}))

inc/mbImgNav.js
/*******************************************************************************
 jquery.mb.components
 Copyright (c) 2001-2010. Matteo Bicocchi (Pupunzi); Open lab srl, Firenze - Italy
 email: info@pupunzi.com
 site: http://pupunzi.com

 Licences: MIT, GPL
 http://www.opensource.org/licenses/mit-license.php
 http://www.gnu.org/licenses/gpl.html
 ******************************************************************************/

/*
 * Name:jquery.mb.imageNavigator
 * Version: 2.0
*/




(function($){
  jQuery.fn.imageNavigator = function (options){
    return this.each (function ()
    {

      this.options={
        areaWidth: 500,
        areaHeight:500,
        defaultnavWidth:150,
        draggerStyle:"2px dotted red",
        navOpacity:.8,
        loaderUrl:"loading.gif",
        additionalContenet:""
      };
      $.extend (this.options, options);

      options= this.options;
      var imgNav= this;

      var additionalContent, draggableElement, applContainer,image,imageW,imageH,imageContainer,imageContainerW,imageContainerH,navLocator,navigationThumb,navigationThumbW,navigationThumbH,nav,navW,navH,image_isHoriz,cont_isHoriz,ratio,actualIdx=0,onScreen=false,navCoordinateX,navCoordinateY;

      $(imgNav).css({
        width : imgNav.options.areaWidth
      });

      var images= $(imgNav).find(".imagesContainer");
      $(imgNav).empty();

      if ($.metadata){
        $.metadata.setType("class");
        $(images).each(function(){
          if ($(this).metadata().imageUrl) $(this).attr("imageUrl",$(this).metadata().imageUrl);
          if ($(this).metadata().navPosition) $(this).attr("navPosition",$(this).metadata().navPosition);
          if ($(this).metadata().navWidth) $(this).attr("navWidth",$(this).metadata().navWidth);
          if ($(this).metadata().NavCoordinates) $(this).attr("NavCoordinates",$(this).metadata().NavCoordinates);
        });
      }

      var imageURL=$(images[0]).attr("imageUrl");
      var navPos=$(images[0]).attr("navPosition");
      var navWidth= $(images[0]).attr("navWidth")?$(this).attr("navWidth"):imgNav.options.defaultnavWidth;//100;

      var titles=$(images).find(".title");
      var descriptions=$(images).find(".description");
      var additionalContents=$(images).find(".additionalContent");


      $(imgNav).append("<div class='imageContainer'></div>");

      imageContainer=$(imgNav).find(".imageContainer");
      $(imageContainer).css({
        overflow:"hidden",
        position: "relative",
        width:this.options.areaWidth+"px",
        height:this.options.areaHeight+"px"
      });
      var loader="<table id='loader' style='display:none;' cellpadding='0' cellspacing='0' width='100%' height='100%'><tr><td valign='middle' align='center'><img src='"+imgNav.options.loaderUrl+"' alt='loading'></td></tr></table>";
      $(imageContainer).append(loader);

      $(imgNav).prepend("<div class='imagesIndex'></div>");
      var imagesIndex=$(imgNav).find(".imagesIndex");
      $(imagesIndex).css({
        position: "relative",
        width:this.options.areaWidth,
        padding: "0"
      });
      $(imagesIndex).append(titles);

      var canClick=true;
      $(titles).each(function(i){
        $(this).click(function(){
          if (!canClick) return;
          actualIdx=i;
          canClick=false;
          imageURL=$(images[i]).attr("imageUrl");
          navPos=$(images[i]).attr("navPosition");
          navWidth= $(images[i]).attr("navWidth")?$(images[i]).attr("navWidth"):imgNav.options.defaultnavWidth;//100;
          if ($(images[i]).attr("NavCoordinates")){
            navCoordinateX=$(images[i]).attr("NavCoordinates").split(",")[0];
            navCoordinateY=$(images[i]).attr("NavCoordinates").split(",")[1];

            //console.log(navCoordinateX,navCoordinateY);
          }

          if(image)
            $(image).fadeOut(500, function() {
              startNav(imageURL);
            });
          else
            startNav(imageURL);
        });
      });

      $(imgNav).append("<div class='descriptionBox'></div>");
      var descriptionBox=$(imgNav).find(".descriptionBox");
      $(descriptionBox).html(descriptions[0]);

      function startNav(u) {
        navW=null;
        ratio=null;
        if (applContainer){
          $(applContainer).remove();
        }
        $(titles).each(function(i){
          if (i==actualIdx)
            $(this).addClass("selected");
          else
            $(this).removeClass("selected");
        });
        navW=navWidth;
        var imageObj = new Image();
        imageObj.src=null;
        u=u+"?rdm="+Math.random();
        imageObj.src = u;
        imageObj.onload =function(){buildnav(u);};
        imageObj.onerror = imageFailed;
        $("#loader").fadeIn(500);
      }

      function imageFailed() {
        alert("non riesco a caricare: "+this.src);
      }

      function buildnav(u) {
        imageContainerW=$(imageContainer).width();
        imageContainerH=$(imageContainer).height();

        $("#loader").fadeOut(500, function(){canClick=true;});
        $(titles).bind("click",function(){return true;});
        $(imageContainer).click(function(){
          if($.browser.msie) $(nav).show();
          else
            $(nav).show();
        });
        $(imageContainer).mouseleave(function(){
          if($.browser.msie) $(nav).hide();
          else
            $(nav).hide();
        });

        //applContainer
        $(imageContainer).append("<div class='applContainer'></div>");
        applContainer = $(imgNav).find(".applContainer");
        $(applContainer).css({
          position:"relative",
          height:$(imageContainer).height()
        });

        $(applContainer).append("<div class='draggableElement'></div>");
        draggableElement=$(applContainer).find(".draggableElement");

        //todo: add a div for image containment

        //image
        $(draggableElement).append("<image class='navImage'>");
        image= $(draggableElement).find(".navImage");

        //additionalContent
        $(draggableElement).append("<div class='additionalContent'></div>");
        additionalContent=$(draggableElement).find(".additionalContent");
        $(additionalContent).css({position:"absolute", top:"0"});
        $(draggableElement).css({
          position:"absolute"
        });

        $(descriptionBox).html(descriptions[actualIdx]);
        $(additionalContent).html(additionalContents[actualIdx]);

        $(image).attr("src",u);
        $(image).hide();
        imageH=$(image).outerHeight();
        imageW=$(image).outerWidth();
        $(image).fadeIn(1000, function(){

          var t,l;
          if (!$(images[actualIdx]).attr("NavCoordinates")){
            t=-(imageH/2-($(nav).height()*3));
            l=-(imageW/2-($(nav).width()*3));
          }else{
            t=-(navCoordinateY-($(imageContainer).height()/2));
            l=-(navCoordinateX-($(imageContainer).width()/2));
            if (t>0) t=0;
            if (l>0) l=0;
          }
          //console.log(!$(images[actualIdx]).attr("NavCoordinates"), t, l)
          $(draggableElement).animate({
            top:t,
            left:l
          },500,"linear");
          refreshThumbPos(l,t);
        });
        $(draggableElement).draggable({
          containment:[$(imageContainer).offset().left-imageW+$(imageContainer).outerWidth(),$(imageContainer).offset().top-imageH+$(imageContainer).outerHeight(),$(imageContainer).offset().left,$(imageContainer).offset().top],
          start:function(){
            $(nav).hide();
            $(draggableElement).css({cursor:"move"});
          },
          stop:function(e,ui){
            refreshThumbPos(ui.position.left,ui.position.top);
            $(draggableElement).css({cursor:"default"});
            $(nav).show();
          }
        });
        $(draggableElement).bind("dblclick", function(){fitonScreen();});

        //nav SCREEN
        $(applContainer).append("<div class='nav'></div>");
        nav = $(imgNav).find(".nav");
        $(nav).css({
          position:"absolute",
          opacity: imgNav.options.navOpacity
        });

        //ZONE SELECTOR
        $(nav).append("<div id='navLocator'></div>");
        navLocator= $(imgNav).find("#navLocator");
        $(navLocator).css({
          zIndex: 10000,
          position: "absolute",
          border : imgNav.options.draggerStyle,
          backgroundColor: $.browser.msie?"white":"transparent",
          opacity: $.browser.msie?.5:1
        });
        $(nav).hide(1);
        $(navLocator).bind("dblclick",function(){fitonScreen();});
        $(navLocator).draggable({
          containment: 'parent',
          start:function(){
            $(navLocator).css({cursor:"move"});
          },
          drag:function(e,ui){
            refreshImagePos(ui.position.left,ui.position.top);
          },
          stop:function(){
            $(navLocator).css({cursor:"default"});
          }
        });

        //THUMB
        $(nav).append("<image class='navigationThumb'>");
        navigationThumb= $(imgNav).find(".navigationThumb");
        $(navigationThumb).attr("src",u);
        $(navigationThumb).bind("dblclick",function(){fitonScreen();});

        image_isHoriz= imageH < imageW;
        cont_isHoriz= imageContainerH < imageContainerW;
        ratio= imageH/navH;
        resetAllValue();
      }

      function fitonScreen(){
        if(!onScreen){
          navLocator.oldX=$(navLocator).css("left");
          navLocator.oldY=$(navLocator).css("top");
          draggableElement.oldX=$(draggableElement).css("left");
          draggableElement.oldY=$(draggableElement).css("top");
          var controller=((navW*imageContainerH)/navH)<imageContainerW;
          if(controller)
            $(image).css("width",imageContainerW);
          else
            $(image).css("height",imageContainerH);
          onScreen=true;
          $(navLocator).css("top",0);
          $(navLocator).css("left",0);
          $(draggableElement).css("top",0);
          $(draggableElement).css("left",0);
          $(draggableElement).bind("mousemove",imgNav.doNothing=function(){return false;});
          $(additionalContent).hide();

        } else{

          $(image).width("");
          $(image).height("");
          onScreen=false;
          $(draggableElement).css("top",draggableElement.oldY);
          $(draggableElement).css("left",draggableElement.oldX);
          $(navLocator).css("top",navLocator.oldY);
          $(navLocator).css("left",navLocator.oldX);
          $(draggableElement).unbind("mousemove",imgNav.doNothing);
          $(additionalContent).show();

        }
        resetAllValue();
        var x=$(navLocator).offsetLeft;
        var y=$(navLocator).offsetTop;
        refreshImagePos(x,y);
      }

      function refreshImagePos(x,y){
        ratio= imageH/navH;
        var posX=-(arguments[0]+1)*ratio;
        var posY=-(arguments [1]+1)*ratio;
        $(draggableElement).css("top",posY);
        $(draggableElement).css("left",posX);
      }

      function refreshThumbPos(x,y){
        ratio= imageH/navH;
        var posX=-(arguments[0]+1)/ratio;
        var posY=-(arguments [1]+1)/ratio;
        $(navLocator).css({
          top:posY,
          left:posX
        });
      }

      function setnavDim(){
        navW=!navW?imageContainerW/4:navW;
        navH=(navW*imageH)/imageW;
        $(navigationThumb).height(parseFloat(navH));
        navigationThumbW=$(navigationThumb).width();
        navigationThumbH=$(navigationThumb).height();
        //			$(nav).css({
        //				overflow:"hidden",
        //				width:navigationThumbW,
        //				height: navigationThumbH+10
        //			})
      }

      function setnavLocatorDim(){
        $(navLocator).width((imageContainerW*navW)/imageW);
        $(navLocator).height((imageContainerH*navH)/imageH);
      }

      function setnavPos(){
        switch(navPos){
          case "TL":
            $(nav).css("left",0);
            $(nav).css("top",0);
            break;
          case "TR":
            $(nav).css("top",0);
            $(nav).css("left",(imageContainerW-navigationThumbW));
            break;
          case "BL":
            $(nav).css("top",(imageContainerH-navigationThumbH));
            break;
          case "BR":
            $(nav).css("left",(imageContainerW-navigationThumbW));
            $(nav).css("top",(imageContainerH-navigationThumbH));
            break;
          default:
            var dim=($(imageContainer).width())-navigationThumbW;
            $(nav).css("left", dim);
            break;
        }
      }

      function resetAllValue(){
        imageContainerW=$(imageContainer).width();
        imageContainerH=$(imageContainer).height();
        cont_isHoriz= imageContainerH < imageContainerW;
        imageH=$(image).height();
        imageW=$(image).width();

        setnavDim();
        setnavLocatorDim();
        setnavPos();

      }

      function fullScreen(){
        if(!image) return;
        $(image).width("");
        $(image).height("");
        if($.browser.msie) $(nav).show();
        else
          $(nav).fadeIn(500);
        imageContainer.oldW=$(imageContainer).css("width");
        imageContainer.oldH=$(imageContainer).css("height");
        imageContainer.style.width= $(window).outerWidth();
        $(window).bind("resize", function(){fullScreen();});
        resetAllValue();
      }
      startNav(imageURL);
    });
  };

  jQuery.fn.extend (
  {
    getMouseX : function (e)
    {
      var mouseX;
      if ($.browser.msie) {
        mouseX = event.clientX + document.body.scrollLeft;
      } else {
        mouseX = e.pageX;
      }
      if (mouseX < 0) {
        mouseX = 0;
      }
      return mouseX;
    },
    getMouseY : function (e)
    {
      var mouseY;
      if ($.browser.msie) {
        mouseY = event.clientY + document.body.scrollTop;
      } else {
        mouseY = e.pageY;
      }
      if (mouseY < 0) {
        mouseY = 0;
      }
      return mouseY;
    }
  });

})(jQuery);
